pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +367,72 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,shootTechnicalID to global getTargetInfo
+*/
+func getTargetMapInfo() {
+	if len(getTargetInfo) == 0 {
+		var target Target
+		ReadTarget(pathTarget, &target)
+		for _, t := range target.Stack() {
+			getTargetInfo[string(t.Kind)] = string(t.Name)
+		}
+
+		shootName := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootName)
+		checkError(err)
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+		// user for case garden-project-shoot add [seed] = seedName
+		if isTargeted("project", "shoot") {
+			getTargetInfo["seed"] = *shoot.Spec.SeedName
+		}
+	}
```

Comment:
nit: I would prefer an early-exit here. This makes the code more readable as you do not have that much nested if-statements
```go
if len(getTargetInfo) > 0 {
  return
}

var target Target
ReadTarget(pathTarget, &target)
for _, t := range target.Stack() {
  getTargetInfo[string(t.Kind)] = string(t.Name)
}

shootName := getTargetInfo["shoot"]
project, err := getProjectObject()
checkError(err)
shoot, err := getShootObject(*project.Spec.Namespace, shootName)
checkError(err)
getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID

// user for case garden-project-shoot add [seed] = seedName
if isTargeted("project", "shoot") {
  getTargetInfo["seed"] = *shoot.Spec.SeedName
}
```

Reply:
fixed